### PR TITLE
DEPR: delim_whitespace kwd in read_csv

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/tests/io/parser/usecols/test_usecols_basic.py [hunk 1]
```diff
@@ -257,13 +257,25 @@ def test_usecols_with_whitespace(all_parsers):
...
+
     if parser.engine == "pyarrow":
         msg = "The 'delim_whitespace' option is not supported with the 'pyarrow' engine"
         with pytest.raises(ValueError, match=msg):
-            parser.read_csv(StringIO(data), delim_whitespace=True, usecols=("a", "b"))
+            with tm.assert_produces_warning(
+                FutureWarning, match=depr_msg, check_stacklevel=False
+            ):
+                parser.read_csv(
+                    StringIO(data), delim_whitespace=True, usecols=("a", "b")
```
Comment (mroeschke) [inline pandas/tests/io/parser/usecols/test_usecols_basic.py:269]: Can most of these be replaced with `sep=r"\s+"` instead already?

File: doc/source/whatsnew/v2.2.0.rst [hunk 2]
```diff
@@ -477,6 +477,7 @@ Other Deprecations
 - Deprecated strings ``H``, ``S``, ``U``, and ``N`` denoting units in :func:`to_timedelta` (:issue:`52536`)
 - Deprecated strings ``H``, ``T``, ``S``, ``L``, ``U``, and ``N`` denoting units in :class:`Timedelta` (:issue:`52536`)
 - Deprecated strings ``T``, ``S``, ``L``, ``U``, and ``N`` denoting frequencies in :class:`Minute`, :class:`Second`, :class:`Milli`, :class:`Micro`, :class:`Nano` (:issue:`52536`)
+- Deprecated the 'delim_whitespace' keyword in :func:`read_csv` and :func:`read_table`, use ``sep="\\s+"`` instead (:issue:`55569`)
```
Comment (mroeschke) [inline doc/source/whatsnew/v2.2.0.rst:480]:
```suggestion
- Deprecated the ``delim_whitespace`` keyword in :func:`read_csv` and :func:`read_table`, use ``sep="\\s+"`` instead (:issue:`55569`)
```

File: pandas/tests/io/parser/usecols/test_usecols_basic.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/tests/io/parser/usecols/test_usecols_basic.py:269]: i think that will introduce a bunch of duplicate tests, this way we can remove these entirely once the deprecation is enforced

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @jbrockmendel

**Pull request merged**